gov2/cross_service/wordfreq/README.md

```diff
@@ -0,0 +1,131 @@
+   -submitterRole "arn:aws:iam::112233445566:role/WordfreqStack-QueueSubmitterRole123EXAM-PLE456789" \
+   ../gutenberg2.txt
+```
+
+you'll see output similar to
+
+```
+2021-08-30T13:28:16.825-0700	DEBUG	uploader/main.go:99	checking for role credentials	{"subRole": "arn:aws:iam::112233445566:role/WordfreqStack-QueueSubmitterRole123EXAM-PLE456789"}
+2021-08-30T13:28:16.826-0700	INFO	uploader/main.go:101	Requesting config from STS	{"rolearn": "arn:aws:iam::112233445566:role/WordfreqStack-QueueSubmitterRole123EXAM-PLE456789"}
+2021-08-30T13:28:17.056-0700	INFO	uploader/main.go:113	Creating new role provider for role	{"roleArn": "arn:aws:iam::112233445566:role/WordfreqStack-QueueSubmitterRole123EXAM-PLE456789"}
+2021-08-30T13:28:17.078-0700	DEBUG	uploader/main.go:123	AssumeRoleProvider can get credentials!	{"tmpAccessKeyID": "AKIAIOSFODNN7EXAMPLE"}
+2021-08-30T13:28:17.078-0700	INFO	uploader/main.go:131	Uploading file to s3	{"bucket": "wordfreqstack-contentbucket-EXAMPLE1-123456789", "key": "08458d49-5c20-4185-a75a-d3ca7eeffe23"}
+2021-08-30T13:28:17.423-0700	INFO	uploader/main.go:153	Uploaded to S3, awaiting results	{"bucket": "wordfreqstack-contentbucket-EXAMPLE1-123456789", "key": "08458d49-5c20-4185-a75a-d3ca7eeffe23"}
+2021-08-30T13:28:19.376-0700	DEBUG	uploader/main.go:204	Got response back:	{"result": "Success"}
+Job Results completed in 16.143640947s for avsdmgqjsdts
+Top Words:
+- which		324
+- paper		234
+- light		212
+- plate		147
+- silver	117
+- picture	114
+- solution	113
+- should	105
+- water		98
+- project	91
+```
+
+## What's happening behind the scenes:
+
+1. The uploader generates a unique random key for its upload, and uploads the file to S3
+2. The S3 bucket, upon the PUT operation completing, places the item onto the SQS queue
+3. The SQS queue delivers the message to the running Fargate container
+4. The Fargate container then processes the message, deletes it, and in turn sends a message to the output queue
+5. The uploader receives the message on the output queue and deletes it.
+6. Any file that fails to process correctly is left for 14 days.
+
+Once the Fargate container receives a message, each object processing step is tracked by another unique ID, a Snowflake. Snowflakes are
+time-stamped unique values pioneered by Twitter. They serve here as a way to track the unique requests that come in in the case of a
+duplicate S3 object key. 
+
+## Limitations
+
+If two clients happen to use the same S3 upload key, their message will be processed in an unpredictable way. At least one uploader will
+receive a success message while the other is likely to get a failure message, since the process deletes items out of S3. Which client will
+get which message is undefined and non-deterministic. If you wish to guarantee that no two clients produce the same key, consider using
+a web service that produces pre-signed S3 URLs for file upload. 
+
+# About the Go CDK
```

Comment:
Go CDK -> AWS CDK for Go

Reply:
I didn't write this, FYI: This is spat out by the CDK.